crates/conjure_core/src/ast/factor.rs

```diff
@@ -15,3 +15,32 @@ pub enum Factor {
     Literal(Literal),
     Reference(Name),
 }
+
+impl std::fmt::Display for Factor {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            Factor::Literal(x) => x.fmt(f),
+            Factor::Reference(Name::UserName(n)) => {
+                write!(f, "{n}")?;
+                Ok(())
+            }
+
+            Factor::Reference(Name::MachineName(x)) => {
+                write!(f, "${x}")?;
```

Comment:
$ is the line comment character in Essence - lets output something like "MN{x}"?

Reply:
what about double underscore, like in C?

Reply:
Done